draft-ietf-quic-multipath.md

```diff
@@ -232,41 +246,82 @@ the use of the multipath extension during the connection handshake,
 
-The enable_multipath parameter MUST NOT be remembered
+The initial_max_paths parameter MUST NOT be remembered
 ({{Section 7.4.1 of QUIC-TRANSPORT}}).
 New paths can only be used after handshake completion.
 
 This extension does not change the definition of any transport parameter
 defined in {{Section 18.2. of QUIC-TRANSPORT}}.
 
 The transport parameter "active_connection_id_limit"
-{{QUIC-TRANSPORT}} limits the number of usable Connection IDs, and also
-limits the number of concurrent paths. However, endpoints might prefer to retain
-spare Connection IDs so that they can respond to unintentional migration events
-({{Section 9.5 of QUIC-TRANSPORT}}).
+{{QUIC-TRANSPORT}} limits the number of usable Connection IDs per path when the
+initial_max_paths parameter is negotiated successfully. 
+Endpoints might prefer to retain spare Connection IDs so that they can 
+respond to unintentional migration events ({{Section 9.5 of QUIC-TRANSPORT}}). 
+
+Endpoints SHOULD use MP_NEW_CONNECTION_ID and MP_RETIRE_CONNECTION_ID 
+frames to provide new Connection IDs for the peer after the initial_max_paths parameter is negotiated. 
+
+Endpoints MUST NOT issue Connection IDs with Path Identifiers larger than 
+the path limitation declared by the initial_max_paths transport parameter 
+and MAX_PATHS frames.
 
 Cipher suites with nonce shorter than 12 bytes cannot be used together with
 the multipath extension. If such cipher suite is selected and the use of the
 multipath extension is negotiated, endpoints MUST abort the handshake with a
 TRANSPORT_PARAMETER error.
 
 
+# Path Identifier {#pathid}
+
+The explicit Path Identifier is an integer between 0 and 2^62 - 1. 
+The Path Identifier is pre-allocated when endpoints provide new Connection IDs
+with MP_NEW_CONNECTION_ID frames {{mp-new-conn-id-frame}}.
+
+Each Connection ID is associated with a Path Identifier, as documented in {{mp-new-conn-id-frame}}. 
+Multiple connection IDs can be associated with the same path identifier.
+
+Endpoints use Path Identifier to address a path in the multipath control frames,
+such as PATH_ABANDON, PATH_STANDBY, and PATH_AVAILABLE frames.
+
+Each endpoint associates a Receiver Packet Number space to each Path Identifier 
+that it provides to the peer. Each endpoint associates a Sender Packet Number space 
+to each Path Identifier received from the peer.
+
+The Path Identifier associated with the Destination Connection ID is used to 
+construct the packet protection nonce defined in {#multipath-aead}.
+
+The Path Identifier associated with the Destination Connection ID 
+is used to identify the path in ACK_MP frames {#ack-mp-frame}.
+
+Note that the Path Identifier for the initial path is 0. Connection IDs
+which are issued by origin NEW_CONNECTION_ID frames {{Section 19.15. of QUIC-TRANSPORT}}
+MUST be treated as their Path Identifier is 0. Also, the Path Identifier for 
+the connection ID specified in the "preferred address" transport parameter is 1.
```

Comment:
Can we treat this as a CID rotation on the initial path? If so, the path ID should remain 0.

Reply:
```suggestion
the connection ID specified in the "preferred address" transport parameter is 0.
Use of the "preferred address" is consider as a migration event that does not change the path ID.
```

Reply:
@qdeconinck @Yanmei-Liu should we commit this change?

Reply:
Just committed.